code/datums/components/defibrillator.dm

```diff
@@ -0,0 +1,382 @@
+			user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Failed to pick up any heart electrical activity.</span>")
+			playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+			busy = FALSE
+			SEND_SIGNAL(parent, COMSIG_DEFIB_SHOCK_APPLIED, user, target, should_cause_harm, FALSE)
+			set_cooldown()
+			return
+
+		var/obj/item/organ/internal/heart/heart = target.get_int_organ(/obj/item/organ/internal/heart)
+		if(heart.status & ORGAN_DEAD)
+			user.visible_message("<span class='boldnotice'>[defib_ref] buzzes: Resuscitation failed - Heart necrosis detected.</span>")
+			playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+			busy = FALSE
+			SEND_SIGNAL(parent, COMSIG_DEFIB_SHOCK_APPLIED, user, target, should_cause_harm, FALSE)
+			set_cooldown()
+			return
+
+		target.set_heartattack(FALSE)
+		SEND_SIGNAL(target, COMSIG_LIVING_MINOR_SHOCK, 100)
+		SEND_SIGNAL(parent, COMSIG_DEFIB_SHOCK_APPLIED, user, target, should_cause_harm, TRUE)
+		set_cooldown()
+		user.visible_message("<span class='boldnotice'>[defib_ref] pings: Cardiac arrhythmia corrected.</span>")
+		target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>", "<span class='userdanger'>You feel a jolt, and your heartbeat seems to steady.</span>")
+		playsound(get_turf(defib_ref), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+		playsound(get_turf(defib_ref), "bodyfall", 50, 1)
+		playsound(get_turf(defib_ref), 'sound/machines/defib_success.ogg', 50, 0)
+		busy = FALSE
+		return
+	if(target.stat != DEAD)
+		user.visible_message("<span class='notice'>[defib_ref] buzzes: Patient is not in a valid state. Operation aborted.</span>")
+		playsound(get_turf(defib_ref), 'sound/machines/defib_failed.ogg', 50, 0)
+		SEND_SIGNAL(parent, COMSIG_DEFIB_ABORTED, user, target, should_cause_harm)
+		busy = FALSE
+		return
+	var/health = target.health
+	target.visible_message("<span class='warning'>[target]'s body convulses a bit.</span>")
+	playsound(get_turf(defib_ref), "bodyfall", 50, 1)
+	playsound(get_turf(defib_ref), 'sound/machines/defib_zap.ogg', 50, 1, -1)
+	for(var/obj/item/organ/external/O in target.bodyparts)
+		total_brute	+= O.brute_dam
+		total_burn	+= O.burn_dam
+	ghost = target.get_ghost(TRUE) // We have to double check whether the dead guy has entered their body during the above
+
+	if(
+		total_burn <= 180 && total_brute <= 180 && \
+		!target.suiciding && !ghost && tplus < tlimit && \
+		!HAS_TRAIT(target, TRAIT_HUSK) && !HAS_TRAIT(target, TRAIT_BADDNA) && \
+		target.blood_volume > BLOOD_VOLUME_SURVIVE && \
+		(target.get_int_organ(/obj/item/organ/internal/heart) || target.get_int_organ(/obj/item/organ/internal/brain/slime)) && \
+		!(signal_result & COMPONENT_BLOCK_DEFIB)
+	)
```

Comment:
split this into multiple ifs please. Making an if statement multiline is generally bad practice and difficult to read (I also understand that this may not be your own work)

Reply:
Yeah, this if statement used to be one very long line and I figured it would be easier to read than what existed beforehand. I can definitely do that, though.